src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,387 @@
+            final PolyLine geometry = new PolyLine(shapePoints);
+            final Map<String, String> tags = protoTagListConverter.convert(protoLine.getTagsList());
+            builder.addLine(identifier, geometry, tags);
+        });
+    }
+
+    private void parseNodes(final PackedAtlasBuilder builder, final List<ProtoNode> nodes)
+    {
+        final ProtoTagListConverter protoTagListConverter = new ProtoTagListConverter();
+        nodes.forEach(protoNode ->
+        {
+            final long identifier = protoNode.getId();
+            final Longitude longitude = Longitude.dm7(protoNode.getLocation().getLongitude());
+            final Latitude latitude = Latitude.dm7(protoNode.getLocation().getLatitude());
+            final Location geometry = new Location(latitude, longitude);
+            final Map<String, String> tags = protoTagListConverter.convert(protoNode.getTagsList());
+            builder.addNode(identifier, geometry, tags);
+        });
+    }
+
+    private void parsePoints(final PackedAtlasBuilder builder, final List<ProtoPoint> points)
+    {
+        final ProtoTagListConverter protoTagListConverter = new ProtoTagListConverter();
+        points.forEach(protoPoint ->
+        {
+            final long identifier = protoPoint.getId();
+            final Longitude longitude = Longitude.dm7(protoPoint.getLocation().getLongitude());
+            final Latitude latitude = Latitude.dm7(protoPoint.getLocation().getLatitude());
+            final Location geometry = new Location(latitude, longitude);
+            final Map<String, String> tags = protoTagListConverter
+                    .convert(protoPoint.getTagsList());
+            builder.addPoint(identifier, geometry, tags);
+        });
+    }
+
+    private RelationBean parseRelationBean(final ProtoRelation protoRelation)
+    {
+        final RelationBean bean = new RelationBean();
+
+        protoRelation.getBeansList().forEach(protoRelationBean ->
+        {
+            final long memberId = protoRelationBean.getMemberId();
+            final String memberRole = protoRelationBean.getMemberRole();
+            final ItemType memberType = ItemType
+                    .forValue(protoRelationBean.getMemberType().getNumber());
+            bean.addItem(memberId, memberRole, memberType);
+        });
+
+        return bean;
+    }
+
+    private void parseRelations(final PackedAtlasBuilder builder,
+            final List<ProtoRelation> relations)
+    {
+        final ProtoTagListConverter protoTagListConverter = new ProtoTagListConverter();
+        relations.forEach(protoRelation ->
+        {
+            final long identifier = protoRelation.getId();
+            final RelationBean bean = parseRelationBean(protoRelation);
+            final Map<String, String> tags = protoTagListConverter
+                    .convert(protoRelation.getTagsList());
+            // TODO should identifier and OSMIdentifier be the same?
```

Comment:
They shouldn't be the same, one is the Atlas identifier, and one is the OSM identifier. Atlas identifier is padded with the country-slicing and way-sectioning digits. We should make a design decision on which one is passed in. This is somewhat tricky - since atlas identifier must be unique per type of entity (a Relation and an Edge can have the same atlas identifier, but no two Relations can have the same atlas identifier). So if this is a true representation of an Atlas, they should pass in the Atlas Identifier and we can leverage our existing factories to figure out the OSM Identifier based on the atlas identifier (remove the 6 digit padding). Let me know if that made sense!

Reply:
Hmm I see. I guess my question is, what should I be passing in for that second parameter? Will I need to have some additional code right here in this function to calculate the correct OSM id? Is there a reason the other builder 'add' methods for Node, Edge, Point, Line, etc. do not have an OSM id parameter?